tests/test_java.py

```diff
@@ -811,3 +813,316 @@ def flush_output_and_stop_async_profiler(self: AsyncProfiledProcess, *args: Any,
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "GCDrainStackTargetSize",
+                    "type": "uintx",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "InitialHeapSize",
+                    "type": "size_t",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MarkStackSize",
+                    "type": "size_t",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MaxHeapSize",
+                    "type": "size_t",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MaxNewSize",
+                    "type": "size_t",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "MinHeapDeltaBytes",
+                    "type": "size_t",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "SegmentedCodeCache",
+                    "type": "bool",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["product"],
+                },
+                {
+                    "name": "UseCompressedClassPointers",
+                    "type": "bool",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["lp64_product"],
+                },
+                {
+                    "name": "UseCompressedOops",
+                    "type": "bool",
+                    "value": None,
+                    "origin": "ergonomic",
+                    "kind": ["lp64_product"],
+                },
+                {"name": "UseG1GC", "type": "bool", "value": None, "origin": "ergonomic", "kind": ["product"]},
+            ],
+        ),
+    ],
+)
+def test_collect_default_jvm_flags(
+    profiler_state: ProfilerState,
+    tmp_path: Path,
+    application_pid: int,
+    expected_flags: List[Dict[str, Union[None, str, List[str]]]],
+) -> None:
+    with make_java_profiler(profiler_state) as profiler:
+        assert_jvm_flags_equal(
+            profiler._metadata.get_jvm_flags_serialized(psutil.Process(application_pid)), expected_flags
+        )
+
+
+@pytest.mark.parametrize(
```

Comment:
No test that I saw so far actually checks the collected value - you use `None` everywhere so it's popped and ignored.

This test seems like a good case to test non-None values, as these are hardcoded that you pass.